strategy/thompson_sampling_strategy.py

```diff
@@ -0,0 +1,152 @@
+    def __init__(self, bandit, alpha_prior, beta_prior):
+        ThompsonSampling.__init__(self, bandit, alpha=alpha_prior, beta=beta_prior)
+
+    def _sample(self, alpha, beta):
+        return scipy.stats.beta.rvs(alpha, beta, size=1)
+
+    def _update_posterior(self, arm_index, observed_reward):
+        if observed_reward == 1:
+            self.posterior_params[arm_index]['alpha'] += 1
+        else:
+            self.posterior_params[arm_index]['beta'] += 1
+
+    @property
+    def estimated_arm_means(self):
+        mean = lambda params: \
+                params['alpha'] / (params['alpha'] + params['beta'])
+        return np.array([ mean(params) for params in self.posterior_params])
+    @property
+    def estimated_arm_sds(self):
+        sd = lambda params: np.sqrt(
+                    (params['alpha'] / (params['alpha'] + params['beta'])) *
+                    (1 - params['alpha'] / (params['alpha'] + params['beta']))
+                )
+        return np.array([sd(params) for params in self.posterior_params])
+
+class ThompsonGaussianKnownSigma(ThompsonSampling):
+    def __init__(self, bandit, sigma, mu_prior, sigma_prior,
+            memory_multiplier=0.9):
+        ThompsonSampling.__init__(self, bandit, mu=mu_prior,
+                sigma2=sigma_prior ** 2)
+        self.sigma2 = sigma ** 2
+        self.sufficient_statistics = [{'n': 0, 'xsum': 0} for _ in range(self.num_arms)]
+        self.memory_multiplier = memory_multiplier
+
+    def _sample(self, mu, sigma2):
+        return np.random.normal(loc=mu, scale=np.sqrt(sigma2))
+
+    def _update_posterior(self, arm_index, observed_reward):
+        sigma2 = self.sigma2
+
+        for index in range(self.num_arms):
+            old_n = self.sufficient_statistics[index]['n']
+            old_xsum = self.sufficient_statistics[index]['xsum']
+            mu_prior = self.prior_params[index]['mu']
+            sigma2_prior = self.prior_params[index]['sigma2']
+
+            if index == arm_index:
+                new_n = self.memory_multiplier * old_n + 1
+                new_xsum = self.memory_multiplier * old_xsum + observed_reward
+            else:
+                new_n = self.memory_multiplier * old_n
+                new_xsum = self.memory_multiplier * old_xsum
+
+            new_sigma2_posterior = 1 / (1 / sigma2_prior + new_n / sigma2)
+            new_mu_posterior = (mu_prior / sigma2_prior + new_xsum / sigma2) * new_sigma2_posterior
+
+            self.sufficient_statistics[index]['n'] = new_n
+            self.sufficient_statistics[index]['xsum'] = new_xsum
+            self.posterior_params[index]['mu'] = new_mu_posterior
+            self.posterior_params[index]['sigma2'] = new_sigma2_posterior
+
+    @property
+    def estimated_arm_means(self):
+        return np.array([params['mu'] for params in self.posterior_params])
+
+    @property
+    def estimated_arm_sds(self):
+        return np.array([self.sigma2 + params['sigma2']
```

Comment:
I think this was in the first commit and fixed in the second. 
Originally, I wanted `estimated_arm_sds` to be the estimated arm standard deviation (posterior predictive standard deviation).  The sum was changed to just `params['sigma2']` in the second commit. 

Don't forget there's a 'view all changes' (so you don't have to look at commits one-by-one)